apps/src/music/blockly/blocks/simple2.js

```diff
@@ -0,0 +1,115 @@
+import {BlockTypes} from '../blockTypes';
+import Globals from '../../globals';
+
+// Examine chain of parents to see if one is 'when_run'.
+const isBlockInsideWhenRun = ctx => {
+  // Since this model doesn't currently support triggered code, then
+  // everything is ultimately called from under when_run.
+  return true;
+};
+
+export const whenRunSimple2 = {
+  definition: {
+    type: BlockTypes.WHEN_RUN_SIMPLE2,
+    message0: 'when run',
+    inputsInline: true,
+    nextStatement: null,
+    colour: 230,
+    tooltip: 'when run',
+    helpUrl: ''
+  },
+  generator: () =>
+    `
+      ProgramSequencer.init();
+      ProgramSequencer.playSequential();
+    `
+};
+
+export const playSoundAtCurrentLocationSimple2 = {
+  definition: {
+    type: BlockTypes.PLAY_SOUND_AT_CURRENT_LOCATION_SIMPLE2,
+    message0: 'play %1',
+    args0: [
+      {
+        type: 'field_sounds',
+        name: 'sound',
+        getLibrary: () => Globals.getLibrary(),
+        playPreview: (id, onStop) => {
+          Globals.getPlayer().previewSound(id, onStop);
+        },
+        currentValue: 'pop/cafe_beat'
+      }
+    ],
+    inputsInline: true,
+    previousStatement: null,
+    nextStatement: null,
+    colour: 230,
+    tooltip: 'play sound',
+    helpUrl: ''
+  },
+  generator: ctx =>
+    `
+      MusicPlayer.playSoundAtMeasureById(
+        "${ctx.getFieldValue('sound')}",
+        ProgramSequencer.getCurrentMeasure(),
+        ${isBlockInsideWhenRun(ctx) ? 'true' : 'false'}
+      );
+      ProgramSequencer.updateMeasureForPlayByLength(
+        MusicPlayer.getLengthForId(
+          "${ctx.getFieldValue('sound')}"
+        )
+      );
+    `
+};
+
+export const playSoundsTogether = {
+  definition: {
+    type: BlockTypes.PLAY_SOUNDS_TOGETHER,
+    message0: 'play together',
+    args0: [],
+    message1: '%1',
+    args1: [
+      {
+        type: 'input_statement',
+        name: 'code'
+      }
+    ],
+    inputsInline: true,
+    previousStatement: null,
+    nextStatement: null,
+    colour: 230,
```

Comment:
This one I'm less sure about, but for now we could use `'music_blocks'`. Any thoughts on how we'd explain the coding construct behind a `play together` block?

Reply:
> Any thoughts on how we'd explain the coding construct behind a play together block?

Great question, and indeed one reason why I was personally hesitant about this model for a long time.  But we decided this looks something like an await for multiple async functions, such as `Promise.all` in JavaScript, `Task.WhenAll` in C#, etc.

Reply:
Still doesn't hurt to try it out and learn from the idea! I'm going to suggest that for now we just give this block the style for `'procedure_blocks`, but that is a very loose opinion. My stronger opinion is just that we shouldn't hard-code colors any more.

Reply:
Added a new `flow_blocks` style and colour, per our offline brainstorming.